asapdiscovery/simulation/scripts/MProFunction.py

```diff
@@ -0,0 +1,239 @@
+import sys
+from pathlib import Path
+import multiprocessing as mp
+
+# Configure logging
+import logging
+from rich.logging import RichHandler
+FORMAT = "%(message)s"
+from rich.console import Console
+logging.basicConfig(
+    level=logging.INFO, format=FORMAT, datefmt="[%X]", handlers=[RichHandler(markup=True)]
+)
+log = logging.getLogger("rich")
+
+## Preliminary Imports
+import os
+import re
+import argparse
+import openmm
+import rdkit
+from rdkit import Chem
+
+## OpenMM Imports
+from openmm.app import PDBFile
+from openmm import unit, app
+from openmm.app import PME, HBonds
+from openmm.app import Modeller
+from openmm import LangevinMiddleIntegrator
+from openmm.app import Simulation
+from openmm import MonteCarloBarostat
+from openmm import Platform
+from openmm.app import DCDReporter
+from openmm.app import StateDataReporter
+from openmm.app import Simulation
+import mdtraj
+from mdtraj.reporters import XTCReporter
+from openff.toolkit.topology import Molecule
+from openmmforcefields.generators import SystemGenerator
+
+#Simulation Parameters
+
+steps = 10000
+equilibrationSteps = 100
+reportingInterval = 1250
+dt = 4.0 * unit.femtoseconds
+temperature = 300 * unit.kelvin
+friction = 1.0 / unit.femtoseconds
+pressure = 1.0 * unit.atmospheres
+barostatInterval = 25
+nonbondedMethod = PME
+nonbondedCutoff = 1.0*unit.nanometers
+ewaldErrorTolerance = 0.000001
+constraints = HBonds
+rigidWater = True
+constraintTolerance = 0.000001
+hydrogenMass = 4.0*unit.amu
+platform = Platform.getPlatformByName('CUDA')
```

Comment:
does this fail at this step if CUDA isn't found? (it would be good if so)

Reply:
Yep! But not as early I'd like it to, I might run again and see where it fails again